### PR TITLE
Add support for Jira fields in issue-config file

### PR DESCRIPTION
**Pull request by kkaarreell**
In issue-config file one can newly set
custom Jira fields. These fields has to
use names as listed at /rest/api/2/field

Example:
```
  - summary: My test epic"
    type: epic
    fields:
      Labels:
       - "CY24Q3"
      "Pool Team": "my_great_team"
      "Story Points": 0
```

### CODE REVIEW

File: newa/__init__.py [hunk 1]
```diff
@@ -1303,11 +1320,43 @@ def create_issue(self,
...
+            for field in fields:
+                field_id = IssueHandler.field_map[field].id_
+                field_type = IssueHandler.field_map[field].type_
+                field_items = IssueHandler.field_map[field].items
+                value = fields[field]
+                # to ease processing set field_values to be always a list of strings
+                if isinstance(value, str):
+                    field_values = [str(value)]
+                elif isinstance(value, list):
+                    field_values = list(map(str, value))
```
Comment (jharuda) [inline newa/__init__.py:1343]: Add support for number type?

**Review by jharuda — CHANGES_REQUESTED** [review]

**Comment by jharuda** [comment]
LGTM.

File: newa/__init__.py (hunk 1 above)
Comment (kkaarreell) [reply newa/__init__.py:1343]: Added.

File: newa/__init__.py [hunk 2]
```diff
@@ -1303,11 +1320,45 @@ def create_issue(self,
...
-                        *jira_issue.fields.labels,
-                        IssueHandler.newa_label]})
+            if fields is None:
+                fields = {}
+            # always add NEWA label to fields
+            if "Labels" in fields and isinstance(fields['Labels'], list):
+                fields['Labels'].append(IssueHandler.newa_label)
+            else:
+                fields['Labels'] = [IssueHandler.newa_label]
+                # IssueHandler.field_map["Pool Team"].id_: [{"value":
```
Comment (The-Mule) [inline newa/__init__.py:1330]: Is this comment still needed in the final version?

**Review by The-Mule — APPROVED** [review]

File: newa/__init__.py (hunk 2 above)
Comment (kkaarreell) [reply newa/__init__.py:1330]: it is not

**Pull request merged**